src/path.rs

```diff
@@ -388,6 +403,62 @@ where
     }
 }
 
+/// Iterator over [`Component`]s in a [`Path`].
+pub struct Components<'a> {
+    path: Path<'a>,
+    offset: usize,
+}
+
+impl<'a> Iterator for Components<'a> {
+    type Item = Component<'a>;
+
+    fn next(&mut self) -> Option<Component<'a>> {
+        let path = &self.path.0;
+
+        if self.offset >= path.len() {
+            return None;
+        }
+
+        if self.offset == 0 && path[0] == Path::SEPARATOR {
+            self.offset += 1;
+            return Some(Component::RootDir);
+        }
+
+        // Coalesce repeated separators like "a//b".
+        while self.offset < path.len() && path[self.offset] == Path::SEPARATOR {
+            self.offset += 1;
+        }
+        if self.offset >= path.len() {
+            return None;
+        }
+
+        let end = if let Some(index) = self
+            .path
+            .0
+            .iter()
+            .skip(self.offset)
+            .position(|b| *b == Path::SEPARATOR)
+        {
+            self.offset + index
+        } else {
+            path.len()
+        };
+
+        let component = &path[self.offset..end];
+        let component = if component == b".." {
+            Component::ParentDir
+        } else {
```

Comment:
What about CurDir?

Reply:
Good catch. I added `CurDir` after I wrote the initial `Components` implementation and forgot to update it. Added it.